PiwikTracker.php

```diff
@@ -1940,6 +1957,62 @@ protected function getCustomVariablesFromCookie()
 
         return json_decode($cookie, $assoc = true);
     }
+
+    /**
+     * Sets a cookie to be sent to the tracking server.
+     *
+     * @param $name
+     * @param $value
+     */
+    public function setOutgoingTrackerCookie($name, $value)
+    {
+        if ($value === null) {
+            unset($this->outgoingTrackerCookies[$name]);
+        }
+        else {
+            $this->outgoingTrackerCookies[$name] = $value;
+        }
+    }
+  
+    /**
+     * Gets a cookie which was set by the tracking server.
+     *
+     * @param $name
+     *
+     * @return bool|string
+     */
+    public function getIncomingTrackerCookie($name)
+    {
+        if (isset($this->incomingTrackerCookies[$name])) {
+            return $this->incomingTrackerCookies[$name];
+        }
+        
+        return false;
+    }
+
+    /**
+     * Reads incoming tracking server cookies.
+     *
+     * @param $cookies Array with HTTP response headers as values
+     */
+    protected function parseIncomingCookies($headers)
+    {
+        $this->incomingTrackerCookies = array();
+        
+        if (!empty($headers)) {
+            foreach($headers as $header) {
+                if (strpos(strtolower($header), 'set-cookie:') !== 0) {
+                    continue;
+                }
+                $cookies = trim(substr($header, 11));
```

Comment:
It's a small thing but I would recommend to use eg `strlen('set-cookie:')` instead of hardcoded `11` as it is a bit better to read/understand (although it is kind of obvious in this case and not so much needed).